corehq/messaging/management/commands/export_conditional_alerts.py

```diff
@@ -0,0 +1,209 @@
+        json_rule.start_date_case_property = action.start_date_case_property
+        json_rule.specific_start_date = action.specific_start_date
+        json_rule.scheduler_module_info = CreateScheduleInstanceActionDefinition.SchedulerModuleInfo(enabled=False)
+
+        return json_rule
+
+    def get_json_scheduling_options(self, schedule):
+        return ExtraSchedulingOptions(
+            active=schedule.active,
+            include_descendant_locations=schedule.include_descendant_locations,
+            default_language_code=schedule.default_language_code,
+            custom_metadata=copy.deepcopy(schedule.custom_metadata),
+            use_utc_as_default_timezone=schedule.use_utc_as_default_timezone,
+            user_data_filter=copy.deepcopy(schedule.user_data_filter),
+            stop_date_case_property_name=schedule.stop_date_case_property_name,
+        )
+
+    def get_json_timed_schedule(self, schedule):
+        if schedule.ui_type != Schedule.UI_TYPE_DAILY:
+            raise CommandError("Expected simple daily schedule")
+
+        json_schedule = SimpleSMSDailyScheduleWithTime(
+            total_iterations=schedule.total_iterations,
+            start_offset=schedule.start_offset,
+            start_day_of_week=schedule.start_day_of_week,
+            repeat_every=schedule.repeat_every,
+            extra_options=self.get_json_scheduling_options(schedule),
+        )
+
+        event = schedule.memoized_events[0]
+        if not isinstance(event, TimedEvent):
+            raise CommandError("Expected TimedEvent")
+
+        json_schedule.time = event.time
+
+        content = event.content
+        if not isinstance(content, SMSContent):
+            raise CommandError("Expected SMSContent")
+
+        json_schedule.message = copy.deepcopy(content.message)
+
+        return json_schedule
+
+    def get_json_alert_schedule(self, schedule):
+        if schedule.ui_type != Schedule.UI_TYPE_IMMEDIATE:
+            raise CommandError("Expected simple immediate schedule")
+
+        json_schedule = SimpleSMSAlertSchedule(
+            extra_options=self.get_json_scheduling_options(schedule),
+        )
+
+        event = schedule.memoized_events[0]
+        content = event.content
+        if not isinstance(content, SMSContent):
+            raise CommandError("Expected SMSContent")
+
+        json_schedule.message = copy.deepcopy(content.message)
+
+        return json_schedule
+
+    def handle(self, domain, **options):
+        result = []
+        for rule in AutomaticUpdateRule.by_domain(
+            domain,
+            AutomaticUpdateRule.WORKFLOW_SCHEDULING,
+            active_only=False,
+        ):
+            json_rule = self.get_json_rule(rule)
+
+            action = rule.memoized_actions[0].definition
+            if action.schedule.location_type_filter:
+                raise CommandError("Expected location_type_filter to be empty for rule %s" % rule.pk)
```

Comment:
Do you think it will be hard for someone who is not intimately familiar with the messaging system to deal with errors like this? Is it expected that the person who is running the command can resolve these errors themselves? For example, identifying the rule by primary key means one has to dig into the database to figure out which rule it was and what to do about the error. Is that a use case you want to address in this PR?

Reply:
all of these errors have to do with the fact that either these are references which need to be resolved in the target domain so it can't be copied directly, or there are rules with use cases that aren't supported yet by the script.

we have a very limited use planned for this right now and the only person doing the exporting is me, but i'll update these error messages to make it more clear what would need to happen to do the exporting in these cases

Reply:
updated export error messages here: https://github.com/dimagi/commcare-hq/pull/21366/commits/c23c9f20e14c5c78e5e47901d772e067b08b1219